.github/workflows/publish.yaml

```diff
@@ -20,56 +20,15 @@ on:
   
-  # Atlas Profile
-  MONGODB_ATLAS_PUBLIC_KEY: ${{ secrets.ATLAS_PUBLIC_KEY_PUBLISHING }}
-  MONGODB_ATLAS_PRIVATE_KEY: ${{ secrets.ATLAS_PRIVATE_KEY_PUBLISHING }}
-  MONGODB_ATLAS_ORG_ID: ${{ secrets.ATLAS_ORG_ID_PUBLISHING }}
-
-  # Organization: atlas dev profile used for organization
-  # currently multi-org-payment-method is set up only in dev atlas account
-  ATLAS_PUBLIC_KEY_DEV: ${{ secrets.ATLAS_PUBLIC_KEY }}
-  ATLAS_PRIVATE_KEY_DEV: ${{ secrets.ATLAS_PRIVATE_KEY }}
-  ATLAS_ORG_ID_DEV: ${{ secrets.ATLAS_ORG_ID }}
-  # currently ATLAS_ORG_OWNER_ID and ATLAS_ORG_OWNER_ID_PUBLISHING also has same value
-  ATLAS_ORG_OWNER_ID: ${{ secrets.ATLAS_ORG_OWNER_ID_PUBLISHING }}
-
-  # federated_settings_org_role_mapping
-  ATLAS_ORG_ID_FOR_FEDERATION: ${{ vars.ATLAS_ORG_ID_FOR_FEDERATION_PUBLISHING }}
-  ATLAS_PUBLIC_KEY_FOR_FEDERATION: ${{ secrets.ATLAS_PUBLIC_KEY_FOR_FEDERATION_PUBLISHING }}
-  ATLAS_PRIVATE_KEY_FOR_FEDERATION: ${{ secrets.ATLAS_PRIVATE_KEY_FOR_FEDERATION_PUBLISHING }}
-  ATLAS_FEDERATED_SETTINGS_ID: ${{ secrets.ATLAS_FEDERATED_SETTINGS_ID_PUBLISHING }}
-
-  # AWS
-  AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID_PUBLISHING}}
-  AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY_PUBLISHING}}
-  AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID_PUBLISHING}}
-
-  # LDAP config
-  LDAP_HOST_NAME: ${{ secrets.LDAP_HOST_NAME_PUBLISHING }}
-  LDAP_BIND_USER_NAME: ${{ secrets.LDAP_BIND_USER_NAME_PUBLISHING }}
-  LDAP_BIND_PASSWORD: ${{ secrets.LDAP_BIND_PASSWORD_PUBLISHING }}
-
-  # third party integration
-  WEBHOOK_CREATE_URL: ${{ secrets.WEBHOOK_CREATE_URL_PUBLISHING }}
-  WEBHOOK_UPDATE_URL: ${{ secrets.WEBHOOK_UPDATE_URL_PUBLISHING }}
-  WEBHOOK_UPDATE_SECRET: ${{ secrets.WEBHOOK_UPDATE_SECRET_PUBLISHING }}
-  PROMETHEUS_USER_NAME: ${{ secrets.PROMETHEUS_USER_NAME_PUBLISHING }}
-  PROMETHEUS_PASSWORD_NAME: ${{ secrets.PROMETHEUS_PASSWORD_NAME_PUBLISHING }}
-  PAGER_DUTY_CREATE_SERVICE_KEY: ${{ secrets.PAGER_DUTY_CREATE_SERVICE_KEY_PUBLISHING }}
-  PAGER_DUTY_UPDATE_SERVICE_KEY: ${{ secrets.PAGER_DUTY_UPDATE_SERVICE_KEY_PUBLISHING }}
-  DATA_DOG_CREATE_API_KEY: ${{ secrets.DATA_DOG_CREATE_API_KEY_PUBLISHING }}
-  DATA_DOG_UPDATE_API_KEY: ${{ secrets.DATA_DOG_UPDATE_API_KEY_PUBLISHING }}
-  OPS_GENIE_API_KEY: ${{ secrets.OPS_GENIE_API_KEY_PUBLISHING }}
-  MICROSOFT_TEAMS_WEBHOOK_CREATE_URL: ${{ secrets.MICROSOFT_TEAMS_WEBHOOK_CREATE_URL_PUBLISHING }}
-  MICROSOFT_TEAMS_WEBHOOK_UPDATE_URL: ${{ secrets.MICROSOFT_TEAMS_WEBHOOK_UPDATE_URL_PUBLISHING }}
-    
 jobs:
   publish:
+    env: # env vars defined here can be referenced in env vars inside the steps
+      MONGODB_ATLAS_BASE_URL: 'https://cloud.mongodb.com/'
+      MONGODB_ATLAS_OUTPUT: 'json'
+      MONGODB_ATLAS_ORG_ID: ${{ github.event.inputs.resourceName=='organization' && secrets.ATLAS_ORG_ID_DEV || github.event.inputs.resourceName=='federated-settings-org-role-mapping' && secrets.ATLAS_ORG_ID_FOR_FEDERATION || secrets.ATLAS_ORG_ID_PUBLISHING }}
```

Comment:
might be worth adding comments here why we use different org_id depending on the resource

Reply:
thanks, adding comment in https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1122/commits/2eccc38b81bc6201d436f863cf4c90f8218e8c08